src/vm/core_test.zig

```diff
@@ -2599,6 +2599,111 @@ test "CairoVM: getFeltRange for non continuous memory" {
     );
 }
 
+test "CairoVM: loadData should give the correct segment size" {
+    // Test setup
+    const allocator = std.testing.allocator;
+    // Create a new VM instance.
+    var vm = try CairoVM.init(allocator, .{});
+    defer vm.deinit();
+    const segment = try vm.segments.addSegment();
+
+    // Prepare data to load into memory
+    var data = std.ArrayList(MaybeRelocatable).init(allocator);
+    defer data.deinit();
+    try data.append(MaybeRelocatable.fromU256(1));
+    try data.append(MaybeRelocatable.fromU256(2));
+    try data.append(MaybeRelocatable.fromU256(3));
+    try data.append(MaybeRelocatable.fromU256(4));
+
+    // Load data into memory segment
+    const actual = try vm.loadData(segment, &data);
+    defer vm.segments.memory.deinitData(std.testing.allocator);
+
+    // Perform assertions
+    try expectEqual(
+        @as(
```

Comment:
why do you need to cast a Relocatable into a Relocatable?

Reply:
Agreed, just fixed